lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -4245,6 +4245,153 @@ SPIRVValue *LLVMToSPIRVBase::transIntrinsicInst(IntrinsicInst *II,
     return BM->addInstTemplate(internal::OpMaskedScatterINTEL, Ops, BB,
                                nullptr);
   }
+  case Intrinsic::is_fpclass: {
+    // There is no direct counterpart for the intrinsic in SPIR-V, hence
+    // we need to emulate its work by sequence of other instructions
+    SPIRVType *ResTy = transType(II->getType());
+    uint64_t FPClass = cast<ConstantInt>(II->getArgOperand(1))->getZExtValue();
+    // if no tests are provided - return false
+    if (FPClass == 0)
+      return BM->addConstant(ResTy, false);
+    // if all tests are provided - return true
+    if (FPClass == fcAllFlags)
+      return BM->addConstant(ResTy, true);
+
+    Type *OpLLVMTy = II->getArgOperand(0)->getType();
+    SPIRVValue *InputFloat = transValue(II->getArgOperand(0), BB);
+    std::vector<SPIRVValue *> ResultVec;
+
+    // Adds test for Negative/Positive values
+    auto GetNegPosInstTest = [&](SPIRVValue *TestInst, bool IsNegative) {
```

Comment:
Hm, I feel like there is room for optimization(s): for example test for sign can be cached at first entry and its ID can be reused by other tests.